drivers/console/Kconfig

```diff
@@ -291,6 +291,13 @@ config NATIVE_POSIX_CONSOLE_INIT_PRIORITY
 	help
 	  Device driver initialization priority.
 
+config SEMIHOST_CONSOLE
+	bool "Use semihosting for console"
+	select CONSOLE_HAS_DRIVER
+	depends on CPU_CORTEX_M
```

Comment:
Why CORTEX_M? This is supported in CORTEX_A too

Reply:
@carlescufi https://github.com/zephyrproject-rtos/zephyr/pull/21073#discussion_r352286180

The current implementation is Cortex-M-specific, and that is arguably where this feature is the most needed since low-end Cortex-M devices are often I/O constrained.

As for Cortex-A and Cortex-R, I plan to add support for them later if there is a need for it (note that the semihosting mechanism can do more than just console I/O, i.e. it can do virtual file I/O as well and this can be useful for Cortex-A and -R as well).

Reply:
Very well, thank you.